### DISCUSSION TITLE
Insert does not seem to do so?

### DISCUSSION THREAD

**Discussion by gudstrand** [msiemens/tinydb#366] [General]
I have a dictionary that defines a vendor, url and some other metadata.  I wanted to create a db of entries that contain the entry append with a notification datetime.    So, I added a datetime to dictionary and after creating a custom datetime serializer I could insert an entry to the db.

The issue is that any attempt to insert a second record for the same vendor, with a different time, results in no change to the database.  What am I doing wrong?

Thanks!
-g

**Comment by msiemens** [comment]
Hey @gudstrand, could you provide an example program that shows the probem you're having? It's a bit difficult to pinpoint the issue without inspecting the code that results in the problem 🙃

**Reply by gudstrand** [reply]
So, I wrote a sample program, and it works. I have no idea why it is failing in my main, but I am back to debugging. Quick question, how and when does tinydb decide to write to the filesystem?

**Reply by msiemens** [reply]
Unless you're using the `CachingMiddleware` it's writing the data on every `insert`/`update`/`remove`/`truncate` operation
